cirq-core/cirq/ops/boolean_hamiltonian.py

```diff
@@ -36,7 +36,20 @@
 
 @value.value_equality
 class BooleanHamiltonianGate(raw_types.Gate):
-    """A gate that represents a Hamiltonian from a set of Boolean functions."""
+    r"""A gate that represents evolution due to a Hamiltonian from a set of Boolean functions.
+
+    This gate constructs a diagonal gate in the computational basis that encodes in its
+    phases classical functions.
+
+    The gate is specified by a list of parameters, $[x_0, x_1, \dots, x_{n-1}]$, a
+    list of boolean expressions that are functions of these parameters,
+    $[f_0(x_0,\dots,x_{n-1}), f_1(x_0,\dots,x_{n-1}), f_{p-1}(x_0,\dots,x_{n-1})]$
```

Comment:
nit: Missing dots between f_1 and f_{p-1}:

```
$[f_0(x_0,\dots,x_{n-1}), f_1(x_0,\dots,x_{n-1}), \dots, f_{p-1}(x_0,\dots,x_{n-1})]$
```

(If it gets too long, you can remove f_1.)